repo/packages/S/spark/13/marathon.json.mustache

```diff
@@ -0,0 +1,117 @@
+{
+    "id": "{{service.name}}",
+    "cpus": {{service.cpus}},
+    "mem": {{service.mem}},
+    "cmd": "/sbin/init.sh",
+    "env": {
+        "DCOS_SERVICE_NAME": "{{service.name}}",
+        "SPARK_HDFS_CONFIG_URL": "{{hdfs.config-url}}",
+        "SPARK_USER": "{{service.user}}",
+        "SPARK_DISPATCHER_MESOS_ROLE": "{{service.role}}",
+        "SPARK_DISPATCHER_MESOS_PRINCIPAL": "{{service.principal}}",
+        "SPARK_DISPATCHER_MESOS_SECRET": "{{service.secret}}",
+{{#security.kerberos.krb5conf}}
+        "SPARK_MESOS_KRB5_CONF_BASE64": "{{security.kerberos.krb5conf}}",
+{{/security.kerberos.krb5conf}}
+{{#security.ssl.enabled}}
+        "SPARK_SSL_ENABLED": "{{security.ssl.enabled}}",
+        "SPARK_SSL_KEYSTOREBASE64": "{{security.ssl.keyStoreBase64}}",
+        "SPARK_SSL_TRUSTSTOREBASE64": "{{security.ssl.trustStoreBase64}}",
+        "SPARK_SSL_KEYSTOREPASSWORD": "{{security.ssl.keyStorePassword}}",
+        "SPARK_SSL_KEYPASSWORD": "{{security.ssl.keyPassword}}",
+        "SPARK_SSL_TRUSTSTOREPASSWORD": "{{security.ssl.keyStorePassword}}",
+        "SPARK_SSL_PROTOCOL": "{{security.ssl.protocol}}",
+        "SPARK_SSL_ENABLEDALGORITHMS": "{{security.ssl.enabledAlgorithms}}",
+{{/security.ssl.enabled}}
+        "ENABLE_HISTORY_SERVER": "{{history-server.enabled}}",
+        "HISTORY_LOG_DIR": "{{history-server.log-dir}}",
+        "HISTORY_CLEANER_ENABLED": "{{history-server.cleaner-enabled}}",
+        "HISTORY_CLEANER_INTERVAL": "{{history-server.cleaner-interval}}",
+        "HISTORY_CLEANER_MAXAGE": "{{history-server.cleaner-maxage}}",
+        "HISTORY_KRB_PRINCIPAL": "{{history-server.kerberos.principal}}",
+        "HISTORY_KRB_KEYTAB": "{{history-server.kerberos.keytab}}",
+{{#security.mesos.authentication.enabled}}
+        "DCOS_SERVICE_ACCOUNT_CREDENTIAL": "file:///run/dcos/etc/mesos/agent_service_account.json",
```

Comment:
Follow @karya0 and @nickbp 's PRs as an example of how to use the new "secrets" json section in Marathon
Or maybe @karya0 will be kind enough to create a PR on top of yours, based on your branch, to modify this to use secrets.